tools/eddystone-validator/EddystoneValidator/app/src/main/java/com/google/sample/eddystonevalidator/BeaconArrayAdapter.java

```diff
@@ -59,48 +59,73 @@ public Beacon getItem(int position) {
     deviceAddress.setText(beacon.deviceAddress);
 
-    TextView uidLabel = (TextView)convertView.findViewById(R.id.uidLabel);
-    TextView uidStatus = (TextView)convertView.findViewById(R.id.uidStatus);
+    TextView rssi = (TextView) convertView.findViewById(R.id.rssi);
+    rssi.setText(String.valueOf(beacon.rssi));
+
+    TextView uidLabel = (TextView) convertView.findViewById(R.id.uidLabel);
+    TextView uidNamespace = (TextView) convertView.findViewById(R.id.uidNamespace);
+    TextView uidInstance = (TextView) convertView.findViewById(R.id.uidInstance);
+    TextView uidTxPower = (TextView) convertView.findViewById(R.id.uidTxPower);
+
+    View uidGroup = convertView.findViewById(R.id.uidGroup);
     if (!beacon.hasUidFrame) {
       grey(uidLabel);
-      uidStatus.setText("no data");
+      uidGroup.setVisibility(View.GONE);
     } else {
       if (beacon.uidStatus.getErrors().isEmpty()) {
         green(uidLabel);
       } else {
         red(uidLabel);
       }
-      uidStatus.setText(beacon.uidStatus.toString());
+      uidNamespace.setText(beacon.uidStatus.uidValue.substring(0, 20));
+      uidInstance.setText(beacon.uidStatus.uidValue.substring(20, 31));
+      uidTxPower.setText(beacon.uidStatus.txPower);
+      uidGroup.setVisibility(View.VISIBLE);
     }
 
-    TextView tlmLabel = (TextView)convertView.findViewById(R.id.tlmLabel);
-    TextView tlmStatus = (TextView)convertView.findViewById(R.id.tlmStatus);
+    TextView tlmLabel = (TextView) convertView.findViewById(R.id.tlmLabel);
+
+    TextView tlmVersion = (TextView) convertView.findViewById(R.id.tlmVersion);
+    TextView tlmVoltage = (TextView) convertView.findViewById(R.id.tlmVoltage);
+    TextView tlmTemp = (TextView) convertView.findViewById(R.id.tlmTemp);
+    TextView tlmAdvCnt = (TextView) convertView.findViewById(R.id.tlmAdvCount);
+    TextView tlmSecCnt = (TextView) convertView.findViewById(R.id.tlmSecCnt);
+
+    View tlmGroup = convertView.findViewById(R.id.tlmGroup);
     if (!beacon.hasTlmFrame) {
       grey(tlmLabel);
-      tlmStatus.setText("not detected");
+      tlmGroup.setVisibility(View.GONE);
     } else {
       if (beacon.tlmStatus.toString().isEmpty()) {
         green(tlmLabel);
-        tlmStatus.setText("OK");
       } else {
         red(tlmLabel);
-        tlmStatus.setText(beacon.tlmStatus.toString());
       }
+      tlmVersion.setText(beacon.tlmStatus.version);
+      tlmVoltage.setText(beacon.tlmStatus.voltage);
+      tlmTemp.setText(beacon.tlmStatus.temp);
+      tlmAdvCnt.setText(beacon.tlmStatus.advCnt);
+      tlmSecCnt.setText(beacon.tlmStatus.secCnt);
+      tlmGroup.setVisibility(View.VISIBLE);
     }
 
-    TextView urlLabel = (TextView)convertView.findViewById(R.id.urlLabel);
-    TextView urlStatus = (TextView)convertView.findViewById(R.id.urlStatus);
+    TextView urlLabel = (TextView) convertView.findViewById(R.id.urlLabel);
+    TextView urlStatus = (TextView) convertView.findViewById(R.id.urlStatus);
```

Comment:
think that diff shows i'd already fixed up the style, but i reran the formatters over everything again just to be sure.